api/go.mod

```diff
@@ -1,8 +1,8 @@
 module go.etcd.io/etcd/api/v3
 
-go 1.22
+go 1.23.0
```

Comment:
Can we keep using 1.23 instead of 1.23.0? If the suffix ".0" is automatically added by `go mod tidy`, then it's OK to keep it.